### ISSUE TITLE
W1028 reports both branches of an Fn::If as unreachable when a condition is used twice in one resource

### ISSUE THREAD

**Issue by zfLQ2qx2** [aws-cloudformation/cfn-lint#4673]
### CloudFormation Lint Version

1.56.0

### What operating system are you using?

Amazon Linux 2023

### Describe the bug

W1028 reports both branches of the same `Fn::If` as unreachable when a condition
is used at two independent places in one resource and the resource schema has a
conditional (`if`/`then`) sub-schema in force.

`cfnlint/rules/functions/IfUnsatisfiable.py` has no logic. The message comes from
`fn_if` in `cfnlint/rules/functions/If.py`, which for each branch does

```python
conditions=validator.context.conditions.evolve({value[0]: True if i == 1 else False})
```

and reports W1028 if that raises `Unsatisfiable`. When the enclosing walk has
already pinned that condition for the scenario it is enumerating, evolving it to
the other value always contradicts, so the check is a tautology rather than a
question about reachability.

The tell is that both branches get reported. In the template below,
`ContainerDefinitions[0]`'s true branch is called unreachable and the nested
`LogConfiguration`'s false branch is called unreachable. `AppLogDriver` is an
unconstrained `String`, so both values of `HasFireLens` are satisfiable and
neither branch is dead.

The two `Fn::If` are siblings, neither nested inside the other. This is not
#3551, where the conditions were layered on the same parameter and the reports
were correct.

What turns it on is the resource schema, not the conditions.
`RequiresCompatibilities: [FARGATE]` selects a conditional sub-schema on
`AWS::ECS::TaskDefinition`; that starts condition-scenario enumeration and the
`Fn::If` nodes get descended again with their condition already pinned. Delete
that one line from the template and both findings go away.

### Expected behavior

No W1028 on the template below. A branch should only be reported when it is
unreachable in every satisfiable scenario, not when it disagrees with the one
scenario currently being enumerated.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Parameters:
  AppLogDriver:
    Type: String
    Default: awslogs
Conditions:
  HasFireLens: !Not [!Equals [!Ref AppLogDriver, 'awslogs']]
Resources:
  TaskDefinition:
    Type: 'AWS::ECS::TaskDefinition'
    Properties:
      NetworkMode: awsvpc
      Cpu: '256'
      Memory: '512'
      RequiresCompatibilities: [FARGATE]
      ContainerDefinitions:
        - !If
          - HasFireLens
          - Name: fluent-bit
            Image: 'amazon/aws-for-fluent-bit:stable'
          - !Ref 'AWS::NoValue'
        - Name: app
          Image: app
          LogConfiguration: !If
            - HasFireLens
            - LogDriver: awsfirelens
            - LogDriver: awslogs
              Options:
                'awslogs-region': us-east-1
                'awslogs-group': g
```

```
$ cfn-lint repro.yaml
W1028 ['Fn::If', 1] is not reachable. When setting condition 'HasFireLens' to True from current status False
repro.yaml:16:7

W1028 ['Fn::If', 2] is not reachable. When setting condition 'HasFireLens' to False from current status True
repro.yaml:27:15
```
